src/diagnostics.rs

```diff
@@ -204,14 +221,16 @@
         write_response_to_fifo(meta, &ctx.diagnostics);
         return;
     }
+    let (_, main_settings) = ctx.language_servers.first_key_value().unwrap();
```

Comment:
if we compute a "most general root path" then we don't need to fall back to the main settings.
That can be done later though